### PR TITLE
plugin/framework/acctests: Add root id attribute requirement information

### PR DESCRIPTION
**Pull request by bflad**
Closes hashicorp/terraform-plugin-framework#80

### CODE REVIEW

File: content/plugin/framework/acctests.mdx [hunk 1]
```diff
@@ -55,5 +55,32 @@ See the [`TestCase`
 documentation](/plugin/sdkv2/testing/acceptance-tests/testcase) for more
 information on using `resource.TestCase`.
 
--> **Note:** We anticipate releasing an improved testing experience for the
-framework in the near future.
+## Implement id Attribute
+
+In SDKv2, resources and data sources automatically included an implicit, root level `id` attribute. The `id` attribute is no longer implicitly added.
```
Comment (bookshelfdave) [inline content/plugin/framework/acctests.mdx:60]: > The `id` attribute is no longer implicitly added.

Are you referring to the SDK or the framework in this statement?

Comment (bflad) [reply content/plugin/framework/acctests.mdx:60]: Ick -- the framework. Let's clarify that!

```suggestion
In SDKv2, resources and data sources automatically included an implicit, root level `id` attribute. In framework, the `id` attribute is no longer implicitly added.
```

File: content/plugin/framework/acctests.mdx [hunk 2]
```diff
@@ -55,5 +55,32 @@ See the [`TestCase`
 documentation](/plugin/sdkv2/testing/acceptance-tests/testcase) for more
 information on using `resource.TestCase`.
 
--> **Note:** We anticipate releasing an improved testing experience for the
-framework in the near future.
+## Implement id Attribute
+
+In SDKv2, resources and data sources automatically included an implicit, root level `id` attribute. In framework, the `id` attribute is no longer implicitly added.
```
Comment (laurapacilio) [inline content/plugin/framework/acctests.mdx:60]:
```suggestion
In SDKv2, resources and data sources automatically included an implicit, root level `id` attribute. In the framework, the `id` attribute is not implicitly added.
```

File: content/plugin/framework/acctests.mdx [hunk 3]
```diff
@@ -7,12 +7,12 @@ description: >-
 
 # Acceptance Tests
 
-The framework for Terraform provider development currently relies on the
-[acceptance test framework](/plugin/sdkv2/testing/acceptance-tests)
-shipped with SDKv2. You will write and run tests with the same steps, and they
-are otherwise indistinguishable from testing against SDKv2.  The major
-difference is in how the provider is specified in [the `TestCase`
-struct](/plugin/sdkv2/testing/acceptance-tests/testcase).
+Testing can be implemented using the [acceptance test framework](/plugin/sdkv2/testing/acceptance-tests) shipped with SDKv2.
```
Comment (laurapacilio) [inline content/plugin/framework/acctests.mdx:10]:
```suggestion
You can implement testing with the [acceptance test framework](/plugin/sdkv2/testing/acceptance-tests) shipped with SDKv2.
```

File: content/plugin/framework/acctests.mdx [hunk 4]
```diff
@@ -7,12 +7,12 @@ description: >-
...
+
+Writing and running tests is similar to SDKv2 providers, with the following exceptions:
+
+- [`TestCase`](/plugin/sdkv2/testing/acceptance-tests/testcase): Specify the provider via [`ProtoV6ProviderFactories`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource#TestCase.ProtoV6ProviderFactories).
```
Comment (laurapacilio) [inline content/plugin/framework/acctests.mdx:14]:
```suggestion
- [`TestCase`](/plugin/sdkv2/testing/acceptance-tests/testcase): Specify the provider with [`ProtoV6ProviderFactories`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource#TestCase.ProtoV6ProviderFactories).
```

**Review by laurapacilio — APPROVED** [review]
A couple wording nits, but otherwise LTGM!

File: content/plugin/framework/acctests.mdx (hunk 3 above)
Comment (bflad) [reply content/plugin/framework/acctests.mdx:10]: I swear, one of these days, I will properly write this type of documentation with active voice! My apologies you always need to nudge me in that direction. 😅

**Pull request merged**